src_features/signTx/ui_flow_signTx.c

```diff
@@ -140,19 +148,23 @@ UX_STEP_NOCB(ux_approval_tx_data_warning_step,
     });
 // clang-format on
 
-UX_FLOW(ux_approval_tx_flow,
-        &ux_approval_tx_1_step,
-        &ux_approval_tx_2_step,
-        &ux_approval_tx_3_step,
-        &ux_approval_tx_4_step,
-        &ux_approval_tx_5_step,
-        &ux_approval_tx_6_step);
+const ux_flow_step_t *ux_approval_tx_flow_[9];
+
+void ux_approve_tx(bool dataPresent) {
+    int step = 0;
+    ux_approval_tx_flow_[step++] = &ux_approval_tx_1_step;
+    if (dataPresent && !N_storage.contractDetails) {
```

Comment:
Notice the `!N_storage.contractDetails`. I copied it from https://github.com/LedgerHQ/app-ethereum/blob/734b200ba16775517da1fd90808a30ee880d10c4/src_features/signTx/logic_signTx.c#L376-L377 but I found it hard to understand.
My initial guess would've been `if (dataPresent && N_storage.contractDetails)`...

Reply:
`contractDetails` is a user setting
`contractDetails` == 1 -> we display all the variables in the contract data
`contractDetails` == 0 -> we just show a warning indicating that the transaction contains some contract interaction

Reply:
Ohhhhh yeah haha I thought `display_warning` was the thing that would ACTUALLY display the variables. My bad !